controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -1577,6 +1578,12 @@ private long getNewEpochMergeCount(final long mergeCountPrevEpoch, ImmutableList
         });
     }
 
+
+    @Override
+    public CompletableFuture<Pair<Map<UUID, TxnStatus>, String>> listCompletedTxns(final int limit, final String continuationToken, final OperationContext context) {
+        return listCompletedTransactions(limit, continuationToken, context);
```

Comment:
instead of invoking abstract method `listCompleteTransactions` here it may be better to defer the implementation of `listCompletedTxns` to subclasses. I don't see the need for one more level of indirection....

Reply:
Done as suggested.